openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2166,6 +2191,96 @@ The following is a non-normative example of a Credential Response containing a C
+
+* in the JOSE header,
+  * `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry [@IANA.JOSE.ALGS]. It MUST NOT be `none` or an identifier for a symmetric algorithm (MAC).
+  * `typ`: REQUIRED. MUST be `keyattestation+jwt`, which explicitly types the key proof JWT as recommended in Section 3.11 of [@!RFC8725].
+
+The key attestation may use `x5c`, `kid` or `trust_chain` (as defined in (#jwt-proof-type) ) to convey the public key and the associated trust mechanism to sign the key attestation.
+
+* in the JWT body,
+  * `iat`: REQUIRED (number). Integer for the time at which the key attestation was issued using the syntax defined in [@!RFC7519].
+  * `exp`: REQUIRED (number). Integer for the time at which the key attestation and the key(s) it is attesting expire, using the syntax defined in [@!RFC7519].
+  * `attested_keys` : REQUIRED. Array of attested keys from the same key storage component using the syntax of JWK as defined in [@!RFC7517].
+  * `key_type` : OPTIONAL. Case sensitive string that asserts the key storage component and its security mechanism of attested keys from the `attested_keys` parameter. This specification defines initial values in (#keyattestation-keytypes).
+  * `user_authentication` : OPTIONAL. Array of case sensitive strings that asserts the security mechanisms the key storage component allows to authenticate the End-User to authorize access to the private key from `keys`. This specification defines initial values in (#keyattestation-auth).
+  * `apr` : OPTIONAL. Array of case sensitive strings that assert attested resistance to specified attack potentials for the given keys. The string values contain URNs that identify the given attack potentials.
+  * `nonce`: OPTIONAL. String that represents a nonce provided by the Issuer to proof that a key attestation was freshly generated.
+  * `status`: OPTIONAL. JSON Object representing the supported revocation check mechanisms, such as the one defined in [@!I-D.ietf-oauth-status-list]
+
+The Credential Issuer MUST validate that the JWT used as a proof is actually signed by a key identified in the JOSE Header.
+
+This is an example of a Key Attestation:
+
+```json
+{
+  "typ": "keyattestation+jwt",
+  "alg": "ES256",
+  "kid": "1"
+}
+.
+{
+  "iss": "<identifier of the issuer of this key attestation>",
+  "iat": 1516247022,
+  "exp": 1541493724,
+  "key_type": "strong_box",
+  "user_authentication": [ "system_pin", "system_biometry" ],
+  "apr" : [ "https://trust-list.eu/apr/high" ],
+  "attested_keys": [
+    {
+      "kty": "EC",
+      "crv": "P-256",
+      "x": "TCAER19Zvu3OHF4j4W4vfSVoHIP1ILilDls7vCeGemc",
+      "y": "ZxjiWWbZMQGHVWKVQ4hbSIirsVfuecCE6t4jT9F2HZQ"
+    }
+  ]
+}
+```
+
+## Key Types {#keyattestation-keytypes}
+
+This specification defines the following values for `key_type`:
+
```

Comment:
Without any further information or references, I believe it is almost impossible for a reader to know what `Trusted Execution Environment` or `Secure Element` mean. If they are supposed to refer to Android and iOS concepts, should we include explicit references or description.

If the OpenID4VCI spec defines these identifiers then it also need to define its meaning in a usable way.

Reply:
agree, references are needed.

Reply:
The best reference I was able to find was section 3.3 of  ISO 27071 (https://www.iso.org/obp/ui/en/#iso:std:iso-iec:27071:ed-1:v1:en) for the general descriptions and then reference ios (secure enclace) and android (strong box) directly?

Reply:
I would say both secure enclave and strong box are subtypes of a secure element.
In general I would say, a secure element is a security chip/component that provides a secure platform that is capable of securely running applications and storing its confidential data (keys etc).

Reply:
Secure Elements usually refers to Common Criteria certified chips that use ISO7816 or ISO14443 from the smartcard industry.

I see that we would use some references and further explanations for the key types

Reply:
i think there were specs for TEE in their own standards body, right?

Reply:
I know this is a rabbit hole but to be even more specific, Secure Elements themselves can also comply to different Common Criteria protection profiles which means they can have different characteristics, i.e., not all of them are equally secure.

Reply:
@awoie that's correct. That's why the `apr` claim is actually the most interesting one, but for some this is hard to measure and some people will not have the security background to know what it actually means, that's why I put key_storage_type and user_authentication as a more approachable/easy way to describe the key storage component